packages/gestalt/src/ButtonLink.js

```diff
@@ -0,0 +1,172 @@
+// @flow strict
+import {
+  type AbstractComponent,
+  forwardRef,
+  Fragment,
+  type Node,
+  useImperativeHandle,
+  useRef,
+} from 'react';
+import getAriaLabel from './accessibility/getAriaLabel.js';
+import NewTabAccessibilityLabel from './accessibility/NewTabAccessibilityLabel.js';
+import { useColorScheme } from './contexts/ColorSchemeProvider.js';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider.js';
+import Flex from './Flex.js';
+import Icon, { type IconColor } from './Icon.js';
+import icons from './icons/index.js';
+import InternalLink from './Link/InternalLink.js';
+import Text from './Text.js';
+
+const DEFAULT_TEXT_COLORS = {
+  blue: 'inverse',
+  gray: 'default',
+  red: 'inverse',
+  transparent: 'default',
+  semiTransparentWhite: 'default',
+  transparentWhiteText: 'inverse',
+  white: 'default',
+};
+
+const SIZE_NAME_TO_PIXEL = {
+  sm: 10,
+  md: 12,
+  lg: 12,
+};
+
+type Target = null | 'self' | 'blank';
+
+type BaseButton = {|
+  accessibilityLabel?: string,
+  color?:
+    | 'gray'
+    | 'red'
+    | 'blue'
+    | 'transparent'
+    | 'semiTransparentWhite'
+    | 'transparentWhiteText'
+    | 'white',
+  dataTestId?: string,
```

Comment:
onClick was removed, bring it back with all the navigation logic

Reply:
DONE